harvester_e2e_tests/integration/test_backup_restore.py

```diff
@@ -0,0 +1,803 @@
+                                 host_shell, vm_shell, need_cleanup,
+                                 ssh_keypair, endpoint_ip):
+        vm_name = restored_vm_name
+        backup_name = vm_backups[1]['metadata']['name']
+        spec = api_client.backups.RestoreSpec.for_new(vm_name)
+        code, data = api_client.backups.restore(backup_name,
+                                                spec)
+        assert 201 == code, (
+            f'Failed to restore backup with new VM, {data}')
+
+        _, private_key = ssh_keypair
+        _wait_for_vm_ready(api_client, wait_timeout, vm_name)
+        ip = _wait_for_vm_ip(endpoint, api_client, vm_name, wait_timeout)
+        with host_shell.login(ipaddr=endpoint_ip, jumphost=True) as host, \
+             vm_shell.login(ipaddr=ip,
+                            username=USERNAME,
+                            password=PASSWORD,
+                            pkey=private_key,
+                            jumphost=host.client) as vm:
+            intact = (_check_file_integrity(vm, 'test-0') and
+                      _check_file_integrity(vm, 'test-1'))
+            assert intact, (
+                'Restore failed, "test-0" file not found or md5 mismatch.')
+
+        if need_cleanup:
+            api_client.vms.delete(vm_name)
+
+    def test_restore_replace_current_vm(self, endpoint, api_client,
+                                        vm_backups, wait_timeout, host_shell,
+                                        vm_shell, need_cleanup, ssh_keypair,
+                                        endpoint_ip, vm_name):
+        backup_name = vm_backups[1]['metadata']['name']
+        spec = api_client.backups.RestoreSpec.for_existing(delete_volumes=True)
+
+        _wait_for_vm_stop(api_client, wait_timeout, vm_name)
+        code, data = api_client.backups.restore(backup_name,
+                                                spec)
+        assert 201 == code, (
+            f'Failed to restore backup with current VM replaced, {data}')
+
+        _, private_key = ssh_keypair
+        _wait_for_vm_ready(api_client, wait_timeout, vm_name)
+        ip = _wait_for_vm_ip(endpoint, api_client, vm_name, wait_timeout)
+        with host_shell.login(ipaddr=endpoint_ip, jumphost=True) as host, \
+             vm_shell.login(ipaddr=ip,
+                            username=USERNAME,
+                            password=PASSWORD,
+                            pkey=private_key,
+                            jumphost=host.client) as vm:
+            intact = (_check_file_integrity(vm, 'test-0') and
+                      _check_file_integrity(vm, 'test-1'))
+            assert intact, (
+                'Restore failed, "test-0" file not found or md5 mismatch.')
+
+        if need_cleanup:
+            api_client.vms.delete(vm_name)
+
+
+@pytest.mark.p0
+@pytest.mark.backupnfs
+class TestDeleteBackupNFS:
+    def test_delete_first_backup(self, endpoint, api_client,

```

Comment:
Before test backup deletion (depends on previous **Test Backup**), we would need setup the environment:
- make a VM having 4 backups
Then we would expected test cases as following (in order):
1. Delete the middle backup and restore the last one
2. Delete the first backup and restore the last one
3. Delete the last backup and restore the last one